sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala

```diff
@@ -137,8 +131,7 @@ private[sql] object FileSourceStrategy extends Strategy with Logging {
           logInfo(s"Planning scan with bin packing, max size: $maxSplitBytes bytes")
 
           val splitFiles = selectedPartitions.flatMap { partition =>
-            partition.files.flatMap { file =>
-              assert(file.getLen != 0, file.toString)
+            partition.files.filter(_.getLen > 0).flatMap { file =>
```

Comment:
Existing CSV test cases covers empty input file. This change is no longer necessary if #11999 is merged.